cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/service/CentralApimlInfoMapper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.service;
+
+import lombok.NonNull;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.cloudgatewayservice.service.model.ApimlInfo;
+import org.zowe.apiml.cloudgatewayservice.service.model.CentralServiceInfo;
+import org.zowe.apiml.services.ServiceInfo;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Strings.emptyToNull;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+
+/**
+ * Build Central Registry Apiml services DTO
+ */
+@Component
+public class CentralApimlInfoMapper {
+    private static final List<String> METADATA_KEYS_WHITELIST = Arrays.asList("zos.sysname", "zos.system", "zos.sysplex", "zos.cpcName", "zos.zosName", "zos.lpar");
```

Comment:
It makes more sense to use Set instead of List. Anyway, what about moving this list into yaml configuration?